### PR TITLE
Add multiple and multiline notes support

### PR DESCRIPTION
**Pull request by Amejonah1200**

### CODE REVIEW

**Comment by Amejonah1200** [comment]
image
---
image

**Review by ISSOtm — CHANGES_REQUESTED** [review]
Perhaps a test could be added?

**Comment by ISSOtm** [comment]
This would be especially useful as `codepan-reporting` has such a feature.

File: src/write.rs [hunk 1]
```diff
@@ -755,13 +755,28 @@ impl<S: Span> Report<'_, S> {
...
-                if !self.config.compact {
-                    write_margin(&mut w, 0, false, false, true, Some((0, false)), &[], &None)?;
-                    write!(w, "\n")?;
+            if is_final_group {
+                for (i, note) in self.notes.iter().enumerate() {
+                    if !self.config.compact {
+                        write_margin(&mut w, 0, false, false, true, Some((0, false)), &[], &None)?;
+                        writeln!(w)?;
+                    }
+                    let note_prefix = format!("{} {}", "Note", i + 1);
```
Comment (zesterer) [inline src/write.rs:764]: A little troubled by the lack of customisation in the message here. I think it's better to have the consumer of the API specify these things by not adding the extra `Note <number>`.

**Review by zesterer — CHANGES_REQUESTED** [review]

File: src/write.rs (hunk 1 above)
Comment (Amejonah1200) [reply src/write.rs:764]: I don't see it part of the PR tbh, as you can't change "Note" in singular form either. So it would be a breaking change...

Comment (ISSOtm) [reply src/write.rs:764]: I think other programs like rustc only print "note" over and over, and IMO that's acceptable behaviour. What do you think?

Comment (Amejonah1200) [reply src/write.rs:764]: It goves a reference point on which note it is and how many there are.

@zesterer what do you think? Keep or not to keep the number?

Comment (zesterer) [reply src/write.rs:764]: I think this is fine on reflection, I managed to misremember how my own crate worked beforehand!

**Review by zesterer — APPROVED** [review]

**Comment by zesterer** [comment]
Just needs a rebase now.

**Comment by Amejonah1200** [comment]
> Perhaps a test could be added?

@ISSOtm Added tests!

**Comment by zesterer** [comment]
Still conflicts, unfortunately :(

**Pull request merged**